src/coreclr/src/gc/unix/gcenv.unix.cpp

```diff
@@ -502,6 +522,27 @@ void GCToOSInterface::FlushProcessWriteBuffers()
     }
     else
     {
+#if defined(TARGET_OSX) && defined(TARGET_ARM64)
+        mach_msg_type_number_t cThreads;
+        thread_act_t *pThreads;
+        kern_return_t machret = task_threads(mach_task_self(), &pThreads, &cThreads);
+        CHECK_MACH("task_threads()", machret);
+
+        uintptr_t sp;
+        uintptr_t registerValues[128];
+
+        // Iterate through each of the threads in the list.
+        for (mach_msg_type_number_t i = 0; i < cThreads; i++)
+        {
+            // Request the threads pointer values to force the thread to emit a memory barrier
+            size_t registers = 128;
```

Comment:
I found the comment:
```
The order and count of the provided register values is unspecified and may change; registers with values that are not valid pointers may be omitted, so the number of pointers returned may vary from call to call.
```

disturbing.  So I just picked a large number. Since it just refers to the size of the stack buffer, didn't think it would make a perf difference..

Reply:
Ok, it makes sense to have it larger then.